sys/vm/vm_page.h

```diff
@@ -889,6 +916,19 @@ vm_page_dirty(vm_page_t m)
 #endif
 }
 
+static __inline void
+vm_page_capdirty(vm_page_t m)
+{
+	vm_page_astate_t mas = vm_page_astate_load(m);
+
+	KASSERT(mas.flags & PGA_CAPSTORE,
+		("vm_page_capdirty w/o PGA_CAPSTORE m=%p", m));
+
+	if ((mas.flags & PGA_CAPDIRTY) == 0) {
```

Comment:
This check confuses me

Reply:
I sort of expect `PGA_CAPDIRTY` to be set already in most `vm_page_capdirty()` calls, just on the grounds that once a page is `PGA_CAPDIRTY` it probably stays that way until laundered or swept by the revoker and found devoid of capabilities, so this load avoids the subsequent `atomic_set_32`, 's all.

Reply:
Are there any potential races with the revoker having cleared the bit on another CPU before this runs and thus failing to set the bit for a new cap dirty write on the current CPU (so the next revocation sweep would potentially miss the write this `vm_page_capdirty` is marking)?

Reply:
That's an excellent question and I can't convince myself that there isn't a race with page reclaim despite that `PGA_CAPDIRTY` is only ever cleared in the laundry or atomically (with `vm_page_aflag_clear`) while the page passes `VM_PAGE_OBJECT_BUSY_ASSERT`.  I'll remove the conditional.